stl/inc/xcharconv_ryu.h

```diff
@@ -429,43 +440,56 @@ inline void __append_d_digits(const uint32_t __olength, uint32_t __digits, char*
     __digits /= 10000;
     const uint32_t __c0 = (__c % 100) << 1;
     const uint32_t __c1 = (__c / 100) << 1;
-    _CSTD memcpy(__result + __olength + 1 - __i - 2, __DIGIT_TABLE + __c0, 2);
-    _CSTD memcpy(__result + __olength + 1 - __i - 4, __DIGIT_TABLE + __c1, 2);
+    _CSTD memcpy(__result + __olength + 1 - __i - 2, __DIGIT_TABLE<char> + __c0, 2);
+    _CSTD memcpy(__result + __olength + 1 - __i - 4, __DIGIT_TABLE<char> + __c1, 2);
     __i += 4;
   }
   if (__digits >= 100) {
     const uint32_t __c = (__digits % 100) << 1;
     __digits /= 100;
-    _CSTD memcpy(__result + __olength + 1 - __i - 2, __DIGIT_TABLE + __c, 2);
+    _CSTD memcpy(__result + __olength + 1 - __i - 2, __DIGIT_TABLE<char> + __c, 2);
     __i += 2;
   }
   if (__digits >= 10) {
     const uint32_t __c = __digits << 1;
-    __result[2] = __DIGIT_TABLE[__c + 1];
+    __result[2] = __DIGIT_TABLE<char>[__c + 1];
     __result[1] = '.';
-    __result[0] = __DIGIT_TABLE[__c];
+    __result[0] = __DIGIT_TABLE<char>[__c];
   } else {
     __result[1] = '.';
     __result[0] = static_cast<char>('0' + __digits);
   }
 }
 
-inline void __append_c_digits(const uint32_t __count, uint32_t __digits, char* const __result) {
+template <class _CharT>
+inline void _Fill_string_zero(_CharT* _Dst, size_t _Size) {
+    if constexpr (is_same_v<_CharT, char>) {
+        memset(_Dst, '0', _Size);

```

Comment:
Could we expand this to other trivial char types and use `_Size * sizeof<_CharT>`?

Reply:
I don't think so, because it's setting the string to the digit '0', not the value 0x00. `memset` sets every byte to the same value, so it can't accomplish, e.g., 0x0030 for UTF-16.

Reply:
Can't it use `wmemset` with `wchar_t`?

Reply:
Why not simply `fill_n(_Dst, _Size, _WIDEN(CharT, '0'))` and let `fill_n` work out the optimization?

Reply:
> Can't it use wmemset with wchar_t?

It sure looks like it, as long as the author is aware of that function's existence 😉 .

> Why not simply fill_n(_Dst, _Size, _WIDEN(CharT, '0')) and let fill_n work out the optimization?

I was hesitant to pull in `<algorithm>` just for that. `fill_n` is also unaware of `wmemset`, so I think @AdamBucior has the better suggestion here.

Reply:
> > Can't it use wmemset with wchar_t?
> 
> It sure looks like it, as long as the author is aware of that function's existence 😉 .
> 
> > Why not simply fill_n(_Dst, _Size, _WIDEN(CharT, '0')) and let fill_n work out the optimization?
> 
> I was hesitant to pull in `<algorithm>` just for that. 

The `fill` family lives in `<xutility>` so we needn't pull in all of `<algorithm>`. 

> `fill_n` is also unaware of `wmemset`, so I think @AdamBucior has the better suggestion here.

IMO it would be more generally useful to teach `fill_n` about `wmemset` than to define this narrowly applicable clone of `fill_n` and teach it about `wmemset`.

Reply:
I suggested using `wmemset` only as code simplification. When you look at it's definition in `<wchar.h>` it's an equivalent loop, so teaching `fill` or `fill_n` about it wouldn't increase performance.
```
_Post_equal_to_(_S)
 _At_buffer_(_S, _Iter_, _N, _Post_satisfies_(_S[_Iter_] == _C))
__inline wchar_t* __CRTDECL wmemset(
    _Out_writes_all_(_N) wchar_t* _S,
    _In_                 wchar_t  _C,
    _In_                 size_t   _N
    )
{
    wchar_t *_Su = _S;
    for (; 0 < _N; ++_Su, --_N)
    {
        *_Su = _C;
    }
    return _S;
}
```

Reply:
`fill_n` is probably simplest of all, then, otherwise I'd just be duplicating its internal logic, as @CaseyCarter pointed out. Thanks for taking a look.